src/python/pants/goal/stats_aggregator.py

```diff
@@ -55,13 +56,32 @@ class StatsAggregatorSubsystem(Subsystem):
         ),
         advanced=True,
     )
+    output_file = StrOption(
+        default=None,
+        metavar="<path>",
+        help="Output the stats to this file. If unspecified, outputs to stdout.",
+    )
+
+
+def _log_or_write_to_file(output_file: Optional[str], text: str) -> None:
+    """Send text to the stdout or write to the output file."""
+    if text:
+        if output_file:
+            with open(output_file, "w") as fh:
```

Comment:
Also, people seem to generally run pants in CI as multiple invocations, e.g. `pants lint check ::` followed by `pants test ::` (https://github.com/pantsbuild/example-python/blob/26d55f2772b72944d7f092019e263f5a5d8984d2/.github/workflows/pants.yaml#L50-L63). It'd be kinda convenient if each invocation could either:

- have its own output file automatically in some way
- be appended to a single output file without overwriting

In particular, it'd be nifty to be able to have a static `pants.ci.toml` that configures this, like:

```toml
[stats]
log = true
output_file = "path/to/file.txt"
```

As it is, it seems that CI that runs multiple separate pants invocations will need to add a unique `--stats-output-file=path/to/file-lint.txt`  to each invocation and then save/export multiple all of those files.

Two ideas:

- open this file in append mode, and print a header about which goal and maybe some metadata like timestamp for each 'write'?
- support some sort of placeholder in the output file name, e.g. `output_file = "path/to/file-{unique}.txt"` that has a unique random value inserted

Reply:
I like the append idea over using unique files.

Reply:
The timestamp and the executed goals are only shown when a file is produced, the console output is unchanged.